GenerateReferenceAssembliesPathTask/GenerateReferenceAssembliesPathTask.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace GenerateReferenceAssembliesPathTask;
+
+// ReSharper disable once UnusedType.Global
+public class GenerateReferenceAssembliesPathTask : Task {
+
+    [Required]
+    // ReSharper disable once MemberCanBePrivate.Global
+    // ReSharper disable once UnusedAutoPropertyAccessor.Global
+    public string? TargetFrameworkRootPath { get; set; }
+
+    [Required]
+    // ReSharper disable once MemberCanBePrivate.Global
+    // ReSharper disable once UnusedAutoPropertyAccessor.Global
+    public string? TargetFrameworkVersion { get; set; }
+
+    [Output]
+    // ReSharper disable once MemberCanBePrivate.Global
+    public string? ClassNameFile { get; set; }
+
+    public override bool Execute() {
+        CreateClass();
+        return !Log.HasLoggedErrors;
+    }
+
+    private void CreateClass() {
+        try {
+            ClassNameFile = "MirrorSharpOptionsWithXmlDocumentation.generated.cs";
+            File.Delete(ClassNameFile);
+
+            // ReSharper disable once ExpressionIsAlwaysNull
+            var settingsClass = $@"namespace MirrorSharp.Tests;
+
+public static partial class MirrorSharpOptionsWithXmlDocumentation {{
+#if NETFRAMEWORK
+    private const string MscorlibReferenceAssemblyPath =
+        @""{TargetFrameworkRootPath}.NETFramework\{TargetFrameworkVersion}\mscorlib.dll"";
+#endif
+}}
+";
```

Comment:
Interesting! I am reluctant to use a whole custom task for this -- if this needs to be written dynamically, would it be simpler to use a built-in `WriteLinesToFile`?

However, the other question is if it needs to be written during build -- is there no way to discover it at runtime? I think .NET Framework is generally easier to discover than modern .NET, without same runtime/SDK distinction.